apis/fluentbit/v1alpha2/plugins/output/splunk_types.go

```diff
@@ -0,0 +1,137 @@
+package output
+
+import (
+	"fmt"
+
+	"github.com/fluent/fluent-operator/apis/fluentbit/v1alpha2/plugins"
+	"github.com/fluent/fluent-operator/apis/fluentbit/v1alpha2/plugins/params"
+)
+
+// +kubebuilder:object:generate:=true
+
+// OpenSearch is the opensearch output plugin, allows to ingest your records into an OpenSearch database.
+type Splunk struct {
+	// IP address or hostname of the target OpenSearch instance, default `127.0.0.1`
+	Host string `json:"host,omitempty"`
+	// TCP port of the target Splunk instance, default `8088`
+	// +kubebuilder:validation:Minimum:=1
+	// +kubebuilder:validation:Maximum:=65535
+	Port *int32 `json:"port,omitempty"`
+	// Specify the Authentication Token for the HTTP Event Collector interface.
+	SplunkToken string `json:"splunk_token,omitempty"`
+	//Buffer size used to receive Splunk HTTP responses: Default `2M`
+	// +kubebuilder:validation:Pattern:="^\\d+(k|K|KB|kb|m|M|MB|mb|g|G|GB|gb)?$"
+	BufferSize string `json:"bufferSize,omitempty"`
+	// Set payload compression mechanism. The only available option is gzip.
+	Compress string `json:"compress,omitempty"`
+	// Specify X-Splunk-Request-Channel Header for the HTTP Event Collector interface.
+	Channel string `json:"channel,omitempty"`
+
+	// Optional username credential for access
+	HTTPUser *plugins.Secret `json:"httpUser,omitempty"`
+	// Password for user defined in HTTP_User
+	HTTPPasswd *plugins.Secret `json:"httpPassword,omitempty"`
+	// If the HTTP server response code is 400 (bad request) and this flag is enabled, it will print the full HTTP request
+	// and response to the stdout interface. This feature is available for debugging purposes.
+	HTTPDebugBadRequest bool `json:"httpDebugBadRequest,omitempty"`
```

Comment:
`*bool` maybe better,like this https://github.com/fluent/fluent-operator/blob/master/apis/fluentbit/v1alpha2/plugins/output/elasticsearch_types.go#L61